middle_end/flambda2/to_cmm/to_cmm_static.ml

```diff
@@ -185,6 +197,37 @@ let static_const0 env res ~updates (bound_static : Bound_static.Pattern.t)
     let float_array = C.emit_float_array_constant sym static_fields in
     let env, res, e = static_float_array_updates sym env res updates 0 fields in
     env, R.update_data res float_array, e
+  | Block_like s, Immutable_int64_array fields ->
+    (* XXX mshinwell: as elsewhere, use custom blocks *)
```

Comment:
This needs to be done and not stay a `XXX` comment: the primitives operating on unboxed arrays will use the helpers from `cmm` which assume that the representation is a custom block, so this current code will very likely produce segfaults and/or memory corruption.
If that code is supposed to be unreachable, then a fatal error would be preferable.